lib/perso.ml

```diff
@@ -1331,6 +1663,41 @@ let get_nb_marriage_witnesses_of_kind fam wk =
     (fun acc (_, w) -> if wk = w then acc + 1 else acc)
     0 witnesses
 
+let number_of_descendants_aux conf base env all_levels sl eval_int =
+  match get_env "level" env with
+  | Vint i -> (
+      match get_env "desc_level_table" env with
+      | Vdesclevtab t ->
+          let m = fst (Lazy.force t) in
+          let cnt =
+            Gwdb.Collection.fold
+              (fun cnt ip ->
+                if all_levels then
+                  if Gwdb.Marker.get m ip <= i then cnt + 1 else cnt
+                else if Gwdb.Marker.get m ip = i then cnt + 1
+                else cnt)
+              0 (Gwdb.ipers base)
+          in
+          VVstring (eval_int conf (if all_levels then cnt - 1 else cnt) sl)
+      | _ -> raise Not_found)
+  | _ -> raise Not_found
+
+let cous_paths_dates_aux conf base env p l1 l2 rev =
```

Comment:
If I understand correctly we use this function to get min/max cousins dates.
The `rev` parameter is a little confusing.
`l1` and `l2` are both strings that are supposed to represent integers, I think it would make more sense to already have them converted to int before calling this function.

I think it will be clearer to have two function `cousins_paths_max_date` and  `cousins_paths_max_date` (that are both wrappers around `cous_paths_dates_aux`, so we can hide away the confusing `rev` parameter )